provider/lib/manager.js

```diff
@@ -257,25 +213,266 @@ module.exports = function (logger, triggerDB, redisClient) {
+				if(!error) {
+
+					resolve(response);
+				} 		
+		    })
+			.catch ( err  => {
+				reject(err); 
+			})			
+		});
+	
+	
+        //********************************************
+        //* try/catch/finally used to handle result of 
+        //* Promise in  await wrappingPostChanges 
+		//* run in endless asynchronous loop 
+		//*********************************************		
+        while( true ){
+  		    try {
+
+                //***********************************************************************
+                //* Before starting to listen on doc changes, check if the trigger id is
+                //* still in the active triggers list. A deleteTrigger() call can delete 
+                //* a trigger id anytime 
+                //***********************************************************************
+                if ( ! self.triggers[triggerData.id]  ) {
+                    //***********************************************
+                    //* end loop for trigger id that does not exist 
+                    //***********************************************
+                    logger.info(method,'Trigger', triggerData.id, " end the listen loop for trigger ");
+                    break;
+                }
+
+                //*************************************************************
+                //* Handling of pause state.   listening on DB changes may be 
+                //* paused for a customerDB on overload situation ( e.g rc=429)
+                //**************************************************************
+                if ( triggerData.pauseState == true ) {
+                    //****************************************************
+                    //* check every 10 sec, if pauseState is still active 
+                    //****************************************************
+                    logger.info(method,'Trigger', triggerData.id, " is in pause_state = true mode ");
+                    await PromiseTimeout(10000);			
+					continue;
+                }
+
+                //TODO: switch off log statement for PROD env  
+                //logger.info(method,'Trigger', triggerData.id, " Wait for " , timeout ,"seconds on a doc change in DB ", dbname);
+                let response = await wrappingPostChanges();	
+                // logger.info(method,'Trigger', triggerData.id, " Received following change from DB = ", response);
+              
+
+                //***********************************************************************
+                //*  if  response =""      -> SDK version < 0.4  returns in json error case an empty string 
+                //*  if  response =  {  "results": [] , "last_seq" : value }  means  timeout occur 
+                //*  if  response = <value>  Doc change received  ( feed.on(data,..))
+                //************************************************************************
+                if ( Object.keys(response).length === 0  ) {
```

Comment:
do we need to check if `response.result` is undefined? the two else paths below rely on this being present

Reply:
In case of the promise.resolve(response) the cloudant SDK ensures that a result element is included:  See API doc
https://cloud.ibm.com/apidocs/cloudant?code=go#postchanges-databases-response